packages/nvidia_nat_autogen/src/nat/plugins/autogen/tool_wrapper.py

```
@@ -0,0 +1,169 @@
+            params: list[inspect.Parameter] = []
+            annotations: dict[str, Any] = {}
+
+            if input_schema is not None:
+                annotations = getattr(input_schema, "__annotations__", {}) or {}
+                for param_name, param_annotation in annotations.items():
+                    resolved_type = resolve_type(param_annotation)
+                    params.append(
+                        inspect.Parameter(param_name, inspect.Parameter.POSITIONAL_OR_KEYWORD,
+                                          annotation=resolved_type))
+                    annotations[param_name] = resolved_type
+            setattr(func_to_wrap, "__signature__", inspect.Signature(parameters=params))
```

Ruff (0.13.2)

Do not call `setattr` with a constant attribute value. It is not any safer than normal property access.

Replace `setattr` with assignment

(B010)